File: digitalmaktabspa/src/models/Schedule.ts
import { ScheduleTime } from "./ScheduleTime";
import { Course } from "./Course";
import { DayOfWeek } from "./DayOfWeek";
import { Teacher } from "./Teacher";

export interface Schedule {
  courseId: string;
  course: Course;
  teacherId: string;
  teacher: Teacher;
  dayOfWeekValue: string;
  dayOfWeek: DayOfWeek;
  scheduleTime: ScheduleTime;
}

File: src/DigitalMaktab.Core.Abstractions/Models/Schedule.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace digitalmaktabapi.Models
{
    [Table("Schedule")]
    public class Schedule : Base
    {
        public required Guid CourseId { get; set; }
        public required Course Course { get; set; }
        public required Guid TeacherId { get; set; }
        public required Teacher Teacher { get; set; }
        public required DayOfWeek DayOfWeek { get; set; }
        public required ScheduleTime ScheduleTime { get; set; }
    }
}